### DISCUSSION TITLE
How to access custom plugins without deploying to npm using Capacitor 3.0

### DISCUSSION THREAD

**Discussion by uncvrd** [ionic-team/capacitor#4390] [Q&A]
Hi there! With the deprecation of accessing custom plugins like so:

```
import { Plugins } from '@capacitor/core'

const { MyCustomPlugin } = Plugins;
```

I was wondering how to access plugins that I do not wish to deploy to npm publicly. For example, I had created a custom authentication plugin `Authenticate` that is to be used for internal purposes only in the company, and (in the past) I was able to extract it for use using object destructuring:

```
const { Authenticate } = Plugins;
```

For my plugin, I simply built out the native code (and registered them in ios/android projects), then since the `Plugins` was an `any` object, I could access my `Authenticate` from the `Plugins` object (without any typings but at least no errors) which worked great.

Will I always need to include the plugin repo as an `npm install` w/ a local file path now?

I guess I'm missing something because this seems like it would be fairly common and my google-fu is failing me, so if anyone can guide me in the best practice for this use case, I'd truly appreciate it!

**Comment by robingenz** [comment]
One option would be to install the plugin directly from a (private) GitHub repository.
For example:
```bash
npm i git+https://github.com/capacitor-community/text-to-speech.git#v0.2.3
```
([capacitor-community/text-to-speech](https://github.com/capacitor-community/text-to-speech)@v0.2.3)

**Comment by netsesame2** [comment]
I have the same issue. 

There are about 5 plugins developed locally, registered in Capacitor v2,  by using Capacitor.Plugins all of them can be invoked very well.
So, no good reason for us to package these local modules into public or private repostories. So please think about our voices,  reserver the "Plugin interface" or something else instead.

Thanks.
